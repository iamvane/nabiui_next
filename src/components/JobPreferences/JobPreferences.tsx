import * as React from 'react';

import {
  Checkbox,
  FormGroup,
  FormControlLabel,
  Typography
} from '@material-ui/core';

import SectionTitle from '../common/SectionTitle';
import {
  JobPreferencesOptions,
  JobPreferencesComponent
} from './constants';
import {
  LessonSizeType,
  AgeGroupType
} from './model';

interface Props extends
  LessonSizeType,
  AgeGroupType {
  handleChange: (event: React.FormEvent<{}>) => void;
}

const JobPreferences: React.StatelessComponent<Props> = props => {
  const JobPreferencesFields = [];

  for (const [key, value] of Object.entries(JobPreferencesOptions)) {
    JobPreferencesFields.push(
      <FormControlLabel
        key={key}
        control={
          <Checkbox
            name={value.name}
            onChange={props.handleChange}
            checked={(props as any)[value.name]}
          />
        }
        label={value.label}
      />
    );
  }

  const JobPreferencesFieldsFirstHalf = JobPreferencesFields.slice(0, 3);
  const JobPreferencesFieldsSecondHalf = JobPreferencesFields.slice(3);
  return (
    <div>
      <SectionTitle text={JobPreferencesComponent.Text.JobPreferences} />
      <Typography className="nabi-margin-top-xsmall nabi-margin-bottom-xsmall">
        {JobPreferencesComponent.Text.SpecifyHowMany}
      </Typography>
      <FormGroup className="nabi-margin-left-small">
        {JobPreferencesFieldsFirstHalf}
      </FormGroup>
      <Typography className="nabi-margin-top-small nabi-margin-bottom-xsmall">
        {JobPreferencesComponent.Text.SpecifyWhatAge}
      </Typography>
      <FormGroup className="nabi-margin-left-small">
        {JobPreferencesFieldsSecondHalf}
      </FormGroup>
    </div>
  );
};

export default (JobPreferences);
