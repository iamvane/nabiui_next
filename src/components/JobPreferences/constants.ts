export enum JobPreferences {
  oneStudent = 'One student',
  smallGroups = 'Small groups (2-4)',
  largeGroups = 'Large groups (+5)',
  children = 'Children (4 to 12 years old)',
  teens = 'Teens (13 to 19 years old)',
  adults = 'Adults (20 to 49 years old)',
  seniors = 'Seniors (50+)'
}

export interface JobPreferencesOptions {
  label: JobPreferences;
  name: string;
}

export const JobPreferencesOptions = {
  Children: <JobPreferencesOptions> {
    label: JobPreferences.children,
    name: 'children'
  },
  Teens: <JobPreferencesOptions> {
    label: JobPreferences.teens,
    name: 'teens'
  },
  Adults: <JobPreferencesOptions> {
    label: JobPreferences.adults,
    name: 'adults'
  },
  Seniors: <JobPreferencesOptions> {
    label: JobPreferences.seniors,
    name: 'seniors'
  },
};

export namespace JobPreferencesComponent {
  export enum Text {
    StudentAge = 'Student Age',
    SpecifyWhatAge = 'Specify what age group you are comfortable working with (select all that apply)',
    SpecifyHowMany = 'Specify how many students you are comfortable teaching at the same time (select all that apply)'
  }
}
