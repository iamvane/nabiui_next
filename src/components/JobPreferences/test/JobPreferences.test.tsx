import * as React from 'react';
import JobPreferences from 'components/JobPreferences/JobPreferences';
import { shallow } from 'enzyme';

describe('JobPreferences', () => {
  let wrapper: any;
  const handleChange: () => void = jest.fn();

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <JobPreferences
          handleChange={handleChange}
          oneStudent={false}
          smallGroups={false}
          largeGroups={false}
          children={false}
          teens={false}
          adults={false}
          seniors={false}
        />
      );
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
