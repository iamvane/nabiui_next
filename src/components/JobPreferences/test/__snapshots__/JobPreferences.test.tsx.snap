// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobPreferences Shallow Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JobPreferences
    adults={false}
    handleChange={[MockFunction]}
    largeGroups={false}
    oneStudent={false}
    seniors={false}
    smallGroups={false}
    teens={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SectionTitle
          text="Job preferences"
        />,
        <WithStyles(Typography)
          className="nabi-margin-top-xsmall nabi-margin-bottom-xsmall"
        >
          Specify how many students you are comfortable teaching at the same time (select all that apply)
        </WithStyles(Typography)>,
        <WithStyles(FormGroup)
          className="nabi-margin-left-small"
        >
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="oneStudent"
                onChange={[MockFunction]}
              />
            }
            label="One student"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="smallGroups"
                onChange={[MockFunction]}
              />
            }
            label="Small groups (2-4)"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="largeGroups"
                onChange={[MockFunction]}
              />
            }
            label="Large groups (+5)"
          />
        </WithStyles(FormGroup)>,
        <WithStyles(Typography)
          className="nabi-margin-top-small nabi-margin-bottom-xsmall"
        >
          Specify what age group are you comfortable working with (select all that apply)
        </WithStyles(Typography)>,
        <WithStyles(FormGroup)
          className="nabi-margin-left-small"
        >
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="children"
                onChange={[MockFunction]}
              />
            }
            label="Children (4 to 12 years old)"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="teens"
                onChange={[MockFunction]}
              />
            }
            label="Teens (13 to 19 years old)"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="adults"
                onChange={[MockFunction]}
              />
            }
            label="Adults (20 to 49 years old)"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="seniors"
                onChange={[MockFunction]}
              />
            }
            label="Seniors (50+)"
          />
        </WithStyles(FormGroup)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "text": "Job preferences",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Specify how many students you are comfortable teaching at the same time (select all that apply)",
          "className": "nabi-margin-top-xsmall nabi-margin-bottom-xsmall",
        },
        "ref": null,
        "rendered": "Specify how many students you are comfortable teaching at the same time (select all that apply)",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="oneStudent"
                  onChange={[MockFunction]}
                />
              }
              label="One student"
            />,
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="smallGroups"
                  onChange={[MockFunction]}
                />
              }
              label="Small groups (2-4)"
            />,
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="largeGroups"
                  onChange={[MockFunction]}
                />
              }
              label="Large groups (+5)"
            />,
          ],
          "className": "nabi-margin-left-small",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "OneStudent",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="oneStudent"
                onChange={[MockFunction]}
              />,
              "label": "One student",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "SmallGroups",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="smallGroups"
                onChange={[MockFunction]}
              />,
              "label": "Small groups (2-4)",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "LargeGroups",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="largeGroups"
                onChange={[MockFunction]}
              />,
              "label": "Large groups (+5)",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Specify what age group are you comfortable working with (select all that apply)",
          "className": "nabi-margin-top-small nabi-margin-bottom-xsmall",
        },
        "ref": null,
        "rendered": "Specify what age group are you comfortable working with (select all that apply)",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="children"
                  onChange={[MockFunction]}
                />
              }
              label="Children (4 to 12 years old)"
            />,
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="teens"
                  onChange={[MockFunction]}
                />
              }
              label="Teens (13 to 19 years old)"
            />,
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="adults"
                  onChange={[MockFunction]}
                />
              }
              label="Adults (20 to 49 years old)"
            />,
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="seniors"
                  onChange={[MockFunction]}
                />
              }
              label="Seniors (50+)"
            />,
          ],
          "className": "nabi-margin-left-small",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Children",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="children"
                onChange={[MockFunction]}
              />,
              "label": "Children (4 to 12 years old)",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Teens",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="teens"
                onChange={[MockFunction]}
              />,
              "label": "Teens (13 to 19 years old)",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Adults",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="adults"
                onChange={[MockFunction]}
              />,
              "label": "Adults (20 to 49 years old)",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Seniors",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="seniors"
                onChange={[MockFunction]}
              />,
              "label": "Seniors (50+)",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SectionTitle
            text="Job preferences"
          />,
          <WithStyles(Typography)
            className="nabi-margin-top-xsmall nabi-margin-bottom-xsmall"
          >
            Specify how many students you are comfortable teaching at the same time (select all that apply)
          </WithStyles(Typography)>,
          <WithStyles(FormGroup)
            className="nabi-margin-left-small"
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="oneStudent"
                  onChange={[MockFunction]}
                />
              }
              label="One student"
            />
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="smallGroups"
                  onChange={[MockFunction]}
                />
              }
              label="Small groups (2-4)"
            />
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="largeGroups"
                  onChange={[MockFunction]}
                />
              }
              label="Large groups (+5)"
            />
          </WithStyles(FormGroup)>,
          <WithStyles(Typography)
            className="nabi-margin-top-small nabi-margin-bottom-xsmall"
          >
            Specify what age group are you comfortable working with (select all that apply)
          </WithStyles(Typography)>,
          <WithStyles(FormGroup)
            className="nabi-margin-left-small"
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="children"
                  onChange={[MockFunction]}
                />
              }
              label="Children (4 to 12 years old)"
            />
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="teens"
                  onChange={[MockFunction]}
                />
              }
              label="Teens (13 to 19 years old)"
            />
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="adults"
                  onChange={[MockFunction]}
                />
              }
              label="Adults (20 to 49 years old)"
            />
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="seniors"
                  onChange={[MockFunction]}
                />
              }
              label="Seniors (50+)"
            />
          </WithStyles(FormGroup)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": "Job preferences",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Specify how many students you are comfortable teaching at the same time (select all that apply)",
            "className": "nabi-margin-top-xsmall nabi-margin-bottom-xsmall",
          },
          "ref": null,
          "rendered": "Specify how many students you are comfortable teaching at the same time (select all that apply)",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(WithFormControlContext(FormControlLabel))
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="oneStudent"
                    onChange={[MockFunction]}
                  />
                }
                label="One student"
              />,
              <WithStyles(WithFormControlContext(FormControlLabel))
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="smallGroups"
                    onChange={[MockFunction]}
                  />
                }
                label="Small groups (2-4)"
              />,
              <WithStyles(WithFormControlContext(FormControlLabel))
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="largeGroups"
                    onChange={[MockFunction]}
                  />
                }
                label="Large groups (+5)"
              />,
            ],
            "className": "nabi-margin-left-small",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "OneStudent",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="oneStudent"
                  onChange={[MockFunction]}
                />,
                "label": "One student",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "SmallGroups",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="smallGroups"
                  onChange={[MockFunction]}
                />,
                "label": "Small groups (2-4)",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "LargeGroups",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="largeGroups"
                  onChange={[MockFunction]}
                />,
                "label": "Large groups (+5)",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Specify what age group are you comfortable working with (select all that apply)",
            "className": "nabi-margin-top-small nabi-margin-bottom-xsmall",
          },
          "ref": null,
          "rendered": "Specify what age group are you comfortable working with (select all that apply)",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(WithFormControlContext(FormControlLabel))
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="children"
                    onChange={[MockFunction]}
                  />
                }
                label="Children (4 to 12 years old)"
              />,
              <WithStyles(WithFormControlContext(FormControlLabel))
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="teens"
                    onChange={[MockFunction]}
                  />
                }
                label="Teens (13 to 19 years old)"
              />,
              <WithStyles(WithFormControlContext(FormControlLabel))
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="adults"
                    onChange={[MockFunction]}
                  />
                }
                label="Adults (20 to 49 years old)"
              />,
              <WithStyles(WithFormControlContext(FormControlLabel))
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="seniors"
                    onChange={[MockFunction]}
                  />
                }
                label="Seniors (50+)"
              />,
            ],
            "className": "nabi-margin-left-small",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Children",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="children"
                  onChange={[MockFunction]}
                />,
                "label": "Children (4 to 12 years old)",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Teens",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="teens"
                  onChange={[MockFunction]}
                />,
                "label": "Teens (13 to 19 years old)",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Adults",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="adults"
                  onChange={[MockFunction]}
                />,
                "label": "Adults (20 to 49 years old)",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Seniors",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="seniors"
                  onChange={[MockFunction]}
                />,
                "label": "Seniors (50+)",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
